### PR TITLE
add activitylog to contract and transfer endpoints

### PR DESCRIPTION
**Pull request by martinhenningjensen**

### CODE REVIEW

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ private async Task DeleteOldTransferAgreementProposals(DateTimeOffset olderThan,
 
         context.TransferAgreementProposals.RemoveRange(oldProposals);
 
+        // TODO: append activity log
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:62]: Missing activitylog

File: libraries/dotnet/EnergyOrigin.ActivityLog/EnergyOrigin.ActivityLog.Unit.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+namespace EnergyOrigin.ActivityLog.Unit.Tests;
+
+public class UnitTest1
+{
+    [Fact]
+    public void Test1()
```
Comment (TopSwagCode) [inline libraries/dotnet/EnergyOrigin.ActivityLog/EnergyOrigin.ActivityLog.Unit.Tests/UnitTest1.cs:6]: Fixed in next interation.

**Review by TopSwagCode — APPROVED** [review]

**Review by TopSwagCode — DISMISSED** [review]

**Review by TopSwagCode — DISMISSED** [review]

File: domains/certificates/Query.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20230101.verified.txt [hunk 3]
```diff
@@ -228,6 +257,98 @@
...
+          "hasMore": {
+            "type": "boolean"
+          }
+        },
+        "additionalProperties": false
+      },
+      "ActorTypeEnum": {
+        "enum": [
+          "User",
+          "System"
```
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/certificates/Query.API/API.IntegrationTests/SwaggerTests.GetSwaggerDocs_v20230101.verified.txt:348]: You need to include activity log in a system. So there is only one system, the one that included it.

File: domains/certificates/Query.API/API/Program.cs [hunk 4]
```diff
@@ -69,14 +70,15 @@
...
-builder.Services.AddDbContext<ApplicationDbContext>(
+builder.Services.AddDbContext<DbContext, ApplicationDbContext>(
     options => options.UseNpgsql(builder.Configuration.GetConnectionString("Postgres")),
     optionsLifetime: ServiceLifetime.Singleton);
 builder.Services.AddDbContextFactory<ApplicationDbContext>();
 
 builder.Services.AddHealthChecks()
     .AddNpgSql(sp => sp.GetRequiredService<IConfiguration>().GetConnectionString("Postgres")!);
 
+builder.Services.AddActivityLog();
```
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/certificates/Query.API/API/Program.cs:81]: I don't see warnings :-(

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs [hunk 5]
```diff
@@ -147,8 +162,18 @@ public class ContractsController : ControllerBase
             NonExistingContract => NotFound(),
             MeteringPointOwnerNoMatch => Forbid(),
             EndDateBeforeStartDate => ValidationProblem("EndDate must be after StartDate"),
-            SetEndDateResult.Success => Ok(),
+            SetEndDateResult.Success => await LogChangedEndDateAndReturnOk(activityLogEntryRepository, user, id),
             _ => throw new NotImplementedException($"{result.GetType()} not handled by {nameof(ContractsController)}")
         };
     }
```
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:168]: It was before, so i guess current functionality should remain

File: domains/certificates/Shared/DataContext/DataContext.csproj [hunk 6]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="EnergyOrigin.ActivityLog" Version="3.0.4" />
```
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/certificates/Shared/DataContext/DataContext.csproj:10]: I dunno, it needs to be bumped every time we change it

File: domains/certificates/Shared/DataContext/Migrations/20240125131645_AddActivityLog.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                name: "ActivityLogs",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    Timestamp = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false),
+                    ActorId = table.Column<Guid>(type: "uuid", nullable: false),
+                    ActorType = table.Column<int>(type: "integer", nullable: false),
+                    ActorName = table.Column<string>(type: "text", nullable: false),
+                    OrganizationTin = table.Column<string>(type: "text", nullable: false),
+                    OrganizationName = table.Column<string>(type: "text", nullable: false),
```
[the remark this answers is not in the document]
Comment (martinhenningjensen) [reply domains/certificates/Shared/DataContext/Migrations/20240125131645_AddActivityLog.cs:24]: Nope, name, organisation and tin are String.Empty when system modifies data.

**Review by sahma19 — DISMISSED** [review]

**Review by sahma19 — APPROVED** [review]

**Pull request merged**